src/content/en/fundamentals/primers/service-workers/a-basic-service-worker.markdown

```diff
@@ -0,0 +1,170 @@
+notes:
+  promises:
+    - "<b>Promises</b>&mdash;Notice the use of <code>.then()</code> at the end of the <code>register()</code> function. This is an example of an ECMAScript 2015 construct called a <a href='http://www.html5rocks.com/en/tutorials/es6/promises/'>promise</a>. Service workers make heavy use of promises. If you've never used promises before, you should familiarize yourself with them before trying to implement a service worker."
+  https-only:
+    - "<b>HTTPS Only</b>&mdash;Service workers can do almost whatever they want to HTTP requests and responses. Since this would make them targets for man-in-the-middle attacks, they must be served over HTTPS. This doesn't mean you need HTTPS for development and testing. Service workers served over localhost will also work."
+  sws-dont-control:
+    - "<b>Service Workers Don't Take Control</b>&mdash;If you look around the web, you'll find that many of the pages discussing service workers refer to the service worker as 'taking control' of a page. But as we saw earlier, a service workers can't change a page's DOM. That's why it may be more accurate to think of a service worker as ready to proxy."
+---
+
+<p class="intro">
+  This section describes the most basic service worker possible. It shows you 
+  how a client gets, or registers, a service worker. It shows you how a service 
+  worker prepares to act as a proxy for its clients. Both subjects have more 
+  depth than is shown in this section. It's not intended to be comprehensive; 
+  It's intended to give you a basic foundation on which to build service worker 
+  knowledge.
+</p>
+
+{% include shared/toc.liquid %}
+
+{% include shared/takeaway.liquid list=page.key-takeaways.tldr %}
+
+## A client registers a service worker
+
+To use a service worker, a client tells the browser to install it by calling
+`register()`. More than one page can implement `register()`. You can implement
+it in either an inline script or a separate JavaScript file. Only the first a
+browser encounters trigger the download. The most basic register implementation
+looks like this:
+
+{% highlight javascript %}
+// Does the browser support service workers?
+if ('serviceWorker' in navigator) {
+  // Yes
+  navigator.serviceWorker.register('/service-worker.js').then(function() {
+    // Actions not covered by this primer.
+  });
+} else {
+  // No, but this client should work anyway.
+};
+{% endhighlight %}
+
+There's little more that needs to be said about clients. Now let's talk about 
+the service worker itself.
+
+{% include shared/note.liquid list=page.notes.promises %}
+
+## Where can the service worker play?
+
+A client tells the service worker where it can operate. This is called the
+service worker's scope, and it's the lowest level of your website where the
+service worker can be used. Let's look at the registration code again, but with
+some changes to the `register()` parameters.
+
+{% highlight javascript %}
```

Comment:
Added this after line 96:

A less than obvious corollary of this that service workers can't be stored in 
an assets folder. For example a registration like this will always fail:

{% highlight javascript %}
register('/assets/js/service-worker.js', {scope: '/'})
{% endhighlight %}

For the scope of this service worker to be `/` it has to be served from 
`/service-worker.js`